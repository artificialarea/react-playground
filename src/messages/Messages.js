import React from 'react';
import './Messages.css';

export default function Messages(props) {
  
  const badge = props.unread ? <div className="unread_count badge">{props.unread}</div> : null;

  return (
    <section className="messages">
      {props.name}
      {badge}
    </section>
  );
}