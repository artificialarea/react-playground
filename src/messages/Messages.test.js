import React from 'react';
import ReactDOM from 'react-dom';
import Messages from './Messages';
import renderer from 'react-test-renderer';

// TEMPLATE (that isn't used in tests below?)
// it('renders without crashing', () => {
//   const input = <some-value>; // e.g. number, string, function(param)
//   const expectedOutput = true; // e.g. Boolean, number, string, etc
//   const actualOutput = Messages(input);
//   expect(actualOutput).toBe(expectedOutput);
// });

describe('<Messages />', () => {

  // SMOKE TEST //////////////////////////////////////////////////////////

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Messages />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // SNAPSHOT TESTS (with 'react-test-renderer') ///////////////////////

  it('renders the UI as expected', () => {
    // Render the component, as JSON
    const tree = renderer
      .create(<Messages name="Messages" unread={4} />)
      .toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/Messages.test.js.snap
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected with no unreads', () => {
    const tree = renderer.create(<Messages name="Messages" unread={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});