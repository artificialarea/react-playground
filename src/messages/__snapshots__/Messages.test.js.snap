// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Messages  renders the UI as expected 1`] = `
<section
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</section>
`;

exports[`<Messages  renders the UI as expected with no unreads 1`] = `
<section
  className="messages"
>
  Messages
</section>
`;

exports[`<Messages /> renders the UI as expected 1`] = `
<section
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</section>
`;

exports[`<Messages /> renders the UI as expected with no unreads 1`] = `
<section
  className="messages"
>
  Messages
</section>
`;

exports[`<Messages component renders the UI as expected 1`] = `
<section
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</section>
`;

exports[`<Messages component renders the UI as expected with no unreads 1`] = `
<section
  className="messages"
>
  Messages
</section>
`;

exports[`Messages component renders the UI as expected 1`] = `
<section
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</section>
`;

exports[`Messages component renders the UI as expected with no unreads 1`] = `
<section
  className="messages"
>
  Messages
</section>
`;
